websocket-codec/src/message.rs

```diff
@@ -160,7 +161,7 @@ impl Decoder for MessageCodec {
             let data = src
                 .split_to(data_range.end)
                 .freeze()
-                .slice(data_range.start, data_range.end);
+                .slice(data_range.start..data_range.end);
```

Comment:
I think this can be simplified to:
```rust
let mut data = src.split_to(data_range.end).split_off(data_range.start);
```

...which gives you a `data: BytesMut`. Then no copying is needed lower down, where you have `buff.put(data);`. (The lines that call `Message::new` need to change to accept `data.freeze()`, but I would not expect `freeze` to copy or allocate).